arelle/Cntlr.py

```diff
@@ -8,18 +8,26 @@
    :license: Apache-2.
    :synopsis: Common controller class to initialize for platform and setup common logger functions
 """
-from arelle import PythonUtil # define 2.x or 3.x string types
+from __future__ import annotations
+from typing import Any, TYPE_CHECKING, TextIO, Mapping
+from .typing import TypeGetText
 import tempfile, os, io, sys, logging, gettext, json, re, subprocess, math
 from arelle import ModelManager
+from arelle.WebCache import WebCache
```

Comment:
I don't know how much drag this stuff has, but should we be putting all of the imports that are just for typing inside of the  `if TYPE_CHECKING` block?

Reply:
Actually this was moved from inside the Cntlr init function, it is needed to initialize WebCache [here](https://github.com/Arelle/Arelle/blob/6e9e22bf1def1d1a7c1411ae53698ede7561e85e/arelle/Cntlr.py#L242)